manifests/init.pp

```diff
@@ -415,9 +417,10 @@
   Enum['ipmitool', 'freeipmi', 'shell'] $bmc_default_provider = $::foreman_proxy::params::bmc_default_provider,
   Boolean $realm = $::foreman_proxy::params::realm,
   Foreman_proxy::ListenOn $realm_listen_on = $::foreman_proxy::params::realm_listen_on,
-  String $realm_provider = $::foreman_proxy::params::realm_provider,
+  Enum['freeipa', 'ad'] $realm_provider = $::foreman_proxy::params::realm_provider,
```

Comment:
I'd prefer to keep this as a string since that allows third party plugins to still use the same infra.

Reply:
Sure, I can tweak that back...